src/ConfigFileParser.cpp

```diff
@@ -0,0 +1,534 @@
+        ip = num;
+    } else
+        ip = value.substr(0, colonIndex);
+
+    if (!isIpAddressValid(ip))
+        throw std::runtime_error("Invalid ip address");
+
+    m_configFile.servers[m_configFile.serverIndex].listen.insert(std::make_pair(ip, 0));
+}
+
+/**
+ * @brief Checks if the value of the listen directive is valid and reads it if that is the case
+ *
+ * The function makes sure that the port is valid in the following ways:
+ * 1. The port must not contain a character other than '0'-'9'
+ * 2. The value of the port must be between 1-65535
+ *
+ * @param port The port to be checked
+ * @return true If the port is valid
+ * @return false If the port is invalid
+ */
+bool ConfigFileParser::isPortValid(const std::string& port) const
+{
+    if (port.find_first_not_of("0123456789") != std::string::npos)
+        return false;
+
+    const int base = 10;
+    const int maxPort = 65535;
+    const int minPort = 1;
+
+    int long portNum = std::strtol(port.c_str(), NULL, base);
+    return !(portNum <= minPort || portNum > maxPort);
+}
+
+/**
+ * @brief Checks if the port is valid and reads it if that is the case
+ *
+ * For the case that no semicolon is can be found in the listen directive, the function checks
+ * if the defined values is valid ip address. If that is true, the function returns.
+ * If that is not true, the function and checks if it is a valid port and reads it.
+ *
+ */
+void ConfigFileParser::readPort(const std::string& value)
+{
+    size_t colonIndex = value.find(':');
+    size_t semicolonIndex = value.find(';');
+    std::string port;
+
+    if (colonIndex == std::string::npos) {
+        std::string num = value.substr(0, semicolonIndex);
+        if (isIpAddressValid(num))
+            return;
+        if (value.find('.') != std::string::npos)
+            throw std::runtime_error("Invalid ip address");
+        port = num;
+    } else
+        port = value.substr(colonIndex + 1, semicolonIndex - colonIndex - 1);
+
+    if (!isPortValid(port))
+        throw std::runtime_error("Invalid port");
+
+    for (std::map<std::string, unsigned short>::iterator it = m_configFile.servers[m_configFile.serverIndex].listen.begin(); it != m_configFile.servers[m_configFile.serverIndex].listen.end(); it++)
+        if (it->second == 0)
+            it->second = std::atoi(port.c_str());
+}
+
+/**
+ * @brief Reads the root path
+ *
+ * The function checks if the root path is valid and reads it if that is the case.
+ *
+ * It makes sure that the path is valid in the following ways:
+ * 1. There is only one root path
+ *
+ * If at the end of the path is a slash, it removes it.
+ *
+ * This function can be used for the server block and location block
+ *
+ * @param block The block which surounds the directive
+ */
+void ConfigFileParser::readRootPath(int block, const std::string& value)
+{
+    std::string rootPath = value;
+
+    if (rootPath.find_first_of("  \t\n\v\f\r") != std::string::npos)
```

Comment:
Maybe disregard, since you need to find position of first whitespace also. 

Instead you could mask the whitespace string with a macro to make it more readable: 
`WHITESPACE = " \t\n\v\f\r"`

Or you use a function in either case. So this one and another.

But i would centralize the fancy whitespace string somewhere :D

Reply:
Nice idea with the macro. Implemented this solution.

Reply:
I think in the meantime we came to the conclusion, that using Preprocessor macros are more a C-style thing. Disadvantages are no type checking, default global scope, harder debugging, other unexpected behavior. I don't think anything would be applicable here, so it's more of a style question.

But it is relatively easy to not use the preprocessor: You could have a `static const char* const whitespace` variable in the class. (note the double const). 
And in cpp define it as `const char* const ConfigFileParser::whitespace = " \t\n\v\f\r"`;
That way its scoped for the class.

Reply:
You are right. We discussed to not use macros anymore. My bad. 

I implemented the approach with the const char* const whitespace variable.